### ISSUE TITLE
 Build local bias track from control step two

### ISSUE THREAD

**Issue by ale07alvarez** [taoliu/MACS#158]
Hi,
I was able to follow this pipeline up to 
Step 4: Build local bias track from control step two

step 1 successful:
macs2 pileup -i INPUTB-N1_sorted.bed -B --extsize 36 -o INPUTB-N1_sorted.bed_d_bg.bdg

output 
INFO  @ Thu, 20 Oct 2016 16:12:45: tag size is determined as 75 bps 
INFO  @ Thu, 20 Oct 2016 16:12:45: # tag size = 75 
INFO  @ Thu, 20 Oct 2016 16:12:45: # total tags in alignment file: 69619356 
INFO  @ Thu, 20 Oct 2016 16:12:45: # Pileup alignment file, extend each read towards up/downstream direction with 36 bps 
INFO  @ Thu, 20 Oct 2016 16:14:32: # Done! Check INPUTB-N1_sorted.bed_d_bg.bdg 

Step 2 The slocal background: successful
macs2 pileup -i INPUTB-N1_sorted.bed -B --extsize 500 -o INPUTB-N1_sorted.bed_1k_bg.bdg

output 
INFO  @ Thu, 20 Oct 2016 16:35:55: tag size is determined as 75 bps 
INFO  @ Thu, 20 Oct 2016 16:35:55: # tag size = 75 
INFO  @ Thu, 20 Oct 2016 16:35:55: # total tags in alignment file: 69619356 
INFO  @ Thu, 20 Oct 2016 16:35:55: # Pileup alignment file, extend each read towards up/downstream direction with 500 bps 
INFO  @ Thu, 20 Oct 2016 16:38:06: # Done! Check INPUTB-N1_sorted.bed_1k_bg.bdg

step3 normalized: not successful
macs2 bdgopt -i INPUTB-N1_sorted.bed_1k_bg.bdg -m multiply -p 0.072 -o INPUTB-N1_sorted_1k_bg_norm.bdg

( fragment length was determined to be 72bp I did 72/1000 = 0.072)

Error:

INFO  @ Thu, 20 Oct 2016 16:57:02: Read and build bedGraph... 
INFO  @ Thu, 20 Oct 2016 17:00:15: Modify bedGraph... 
Traceback (most recent call last):
  File "/share/apps/python2.7.6/bin/macs2", line 5, in <module>
    pkg_resources.run_script('MACS2==2.1.0.20150731', 'macs2')
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/distribute-0.6.28-py2.7.egg/pkg_resources.py", line 499, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/distribute-0.6.28-py2.7.egg/pkg_resources.py", line 1239, in run_script
    execfile(script_filename, namespace, namespace)
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/MACS2-2.1.0.20150731-py2.7-linux-x86_64.egg/EGG-INFO/scripts/macs2", line 616, in <module>
    main()
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/MACS2-2.1.0.20150731-py2.7-linux-x86_64.egg/EGG-INFO/scripts/macs2", line 78, in main
    run( args )  
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/MACS2-2.1.0.20150731-py2.7-linux-x86_64.egg/MACS2/bdgopt_cmd.py", line 58, in run
    btrack.apply_func( lambda x: x \* options.extraparam)
  File "MACS2/IO/BedGraph.pyx", line 830, in MACS2.IO.BedGraph.bedGraphTrackI.apply_func (MACS2/IO/BedGraph.c:11540)
TypeError: 'float' object has no attribute '**getitem**'
[alerodri@compute-0-15 macs2_advancedresults]$ /share/apps/python2.7.6/bin/macs2 bdgopt -i /home/alerodri/ChIPSeq-RORA/sam_UCSC/macs2_advancedresults/combined-C-INPUTB-N1_sorted.bed_1k_bg.bdg -m multiply -p 0.072 -o /home/alerodri/ChIPSeq-RORA/sam_UCSC/macs2_advancedresults/combined-C-INPUTB-N1_sorted_1k_bg_norm.bdg
INFO  @ Thu, 20 Oct 2016 17:05:23: Read and build bedGraph... 
INFO  @ Thu, 20 Oct 2016 17:08:39: Modify bedGraph... 
Traceback (most recent call last):
  File "/share/apps/python2.7.6/bin/macs2", line 5, in <module>
    pkg_resources.run_script('MACS2==2.1.0.20150731', 'macs2')
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/distribute-0.6.28-py2.7.egg/pkg_resources.py", line 499, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/distribute-0.6.28-py2.7.egg/pkg_resources.py", line 1239, in run_script
    execfile(script_filename, namespace, namespace)
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/MACS2-2.1.0.20150731-py2.7-linux-x86_64.egg/EGG-INFO/scripts/macs2", line 616, in <module>
    main()
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/MACS2-2.1.0.20150731-py2.7-linux-x86_64.egg/EGG-INFO/scripts/macs2", line 78, in main
    run( args )  
  File "/share/apps/python2.7.6/lib/python2.7/site-packages/MACS2-2.1.0.20150731-py2.7-linux-x86_64.egg/MACS2/bdgopt_cmd.py", line 58, in run
    btrack.apply_func( lambda x: x \* options.extraparam)
  File "MACS2/IO/BedGraph.pyx", line 830, in MACS2.IO.BedGraph.bedGraphTrackI.apply_func (MACS2/IO/BedGraph.c:11540)
TypeError: 'float' object has no attribute '**getitem**'

Please help,
thanks!

**Comment by taoliu** [comment]
Hi @ale07alvarez, could you test it on the latest version 2.1.1.20160309? I haven't found such error while I try 'bdgopt -m multiply -p 0.xxx'

**Comment by passdan** [comment]
I am getting the same error:

```
daniel-desktop:~/Projects/ALD/totalcoverage/lt120/macs2$ macs2 bdgopt -i d_1k_10k_bg_norm.bdg -m multiply -p 0.07537513 -o local_bias_raw.bdg
INFO  @ Tue, 24 Jan 2017 12:01:32: Read and build bedGraph... 
INFO  @ Tue, 24 Jan 2017 12:02:09: Modify bedGraph... 
Traceback (most recent call last):
  File "/usr/bin/macs2", line 614, in <module>
    main()
  File "/usr/bin/macs2", line 76, in main
    run( args )        
  File "/usr/lib/python2.7/dist-packages/MACS2/bdgopt_cmd.py", line 58, in run
    btrack.apply_func( lambda x: x * options.extraparam)
  File "MACS2/IO/BedGraph.pyx", line 830, in MACS2.IO.BedGraph.bedGraphTrackI.apply_func (MACS2/IO/BedGraph.c:11939)
TypeError: 'float' object has no attribute '__getitem__'
```

**Comment by passdan** [comment]
Confirmed fixed in MACS2-2.1.1.20160309 and numpy-1.12.0

**Comment by taoliu** [comment]
Thanks for reporting back!
